### PR TITLE
Speed up boot times and Improve Json (De)Serialization performance and reduce memory usage by reusing JsonSerializerSettings

### PR DESCRIPTION
**Pull request by nzdev**
### Prerequisites

- [x] I have added steps to test this contribution in the description below

If there's an existing issue for this PR then this fixes

### Description
What?
Reduce the number of instantiations of JsonSerializerSettings.

Why?
Improve performance and reduce memory allocations. 
This is especially important in the DatabaseDataSource class as the number of JsonSerializerSettings instantiations at boot is (number of content nodes) *2 + (number of media nodes) + (number of member nodes).

```
//                                Method |         Mean |        Error |      StdDev |  Ratio | RatioSD | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
//-------------------------------------- |-------------:|-------------:|------------:|-------:|--------:|------------:|------------:|------------:|--------------------:|
//   SerializerSettingsInstantiation1000 | 29,120.48 ns | 5,532.424 ns | 303.2508 ns | 997.84 |   23.66 |     73.8122 |           - |           - |            232346 B |
// SerializerSettingsSingleInstantiation |     29.19 ns |     8.089 ns |   0.4434 ns |   1.00 |    0.00 |      0.0738 |           - |           - |               232 B |
```

How?
Determine where JsonSerializerSettings can be reused and instantiate only once.

How to test?

Existing tests pass.

### CODE REVIEW

File: src/Umbraco.Web/Mvc/JsonNetResult.cs [hunk 1]
```diff
@@ -22,10 +22,19 @@ public class JsonNetResult : ActionResult
...
         public Formatting Formatting { get; set; }
 
+        /// <summary>
+        /// Default, unchanged JsonSerializerSettings
+        /// </summary>
+        public static readonly JsonSerializerSettings DefaultJsonSerializerSettings = new JsonSerializerSettings();
+
         public JsonNetResult()
         {
             SerializerSettings = new JsonSerializerSettings();
```
Comment (Shazwazza) [inline src/Umbraco.Web/Mvc/JsonNetResult.cs:32]: Should this change to `SerializerSettings  = DefaultJsonSerializerSettings` ?

**Review by Shazwazza — APPROVED** [review]
Left one comment in there. 

Also Yes! 10000%. I'll add tags and ping @nul800sebastiaan in hopes we can get this reviewed/tested/merged for next minor.

File: src/Umbraco.Web/Mvc/JsonNetResult.cs (hunk 1 above)
Comment (nzdev) [reply src/Umbraco.Web/Mvc/JsonNetResult.cs:32]: Not if you change the value of DefaultJsonSerializerSettings.

**Comment by nul800sebastiaan** [comment]
Thanks @nzdev - makes total sense! 👍

**Pull request merged**